lualib/resty/events/worker.lua

```diff
@@ -271,15 +286,32 @@ function _M:communicate()
 
     wait(read_thread_co)
     wait(write_thread_co)
-    wait(events_thread_co)
 
-    start_timer(self, random_delay())
+    start_communicate_timer(self, random_delay())
+end
+
+function _M:process_events()
+    local events_thread_co = spawn(events_thread, self)
+    local ok, err, perr = wait(events_thread_co)
+    if exiting() then
+        return
+    end
+
+    if not ok then
+        log(ERR, "event worker failed: ", err)
+    end
+
+    if perr then
+        log(ERR, "event worker failed: ", perr)
+    end
+
+    start_process_events_timer(self, random_delay())
```

Comment:
Should the delay be `0`? I think that for events processing we need not to wait.

Reply:
@chronolaw thanks, changing.

Reply:
@chronolaw fixed.